### PR TITLE
VBE detection enhancement

### PR DESCRIPTION
**Pull request by valentinbreiz**
https://github.com/CosmosOS/Cosmos/issues/1383

### CODE REVIEW

File: source/Cosmos.HAL2/Drivers/Video/VBEDriver.cs [hunk 1]
```diff
@@ -76,19 +76,35 @@ private enum EnableValues
...
+            PCIDevice videocard;
+
+            if (VBE.IsAvailable()) //VBE VESA Enabled Mulitboot Parsing
             {
                 Global.mDebugger.SendInternal($"Creating VBE VESA driver with Mode {xres}*{yres}@{bpp}");
                 IO.LinearFrameBuffer = new MemoryBlock(VBE.getLfbOffset(), (uint)xres * yres * (uint)(bpp / 8));
                 lastbuffer = new ManagedMemoryBlock((uint)xres * yres * (uint)(bpp / 8));
             }
-            else
+            else if (Available()) //Bochs Graphics Adaptor ISA Mode
```
Comment (quajak) [inline source/Cosmos.HAL2/Drivers/Video/VBEDriver.cs:87]: Can you rename the method so the comment information is contained?

File: source/Cosmos.HAL2/PCIDevice.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using Cosmos.Core.IOGroup;
 using Cosmos.Common.Extensions;
 using Cosmos.Debug.Kernel;
+using System.Threading;
```
Comment (quajak) [inline source/Cosmos.HAL2/PCIDevice.cs:8]: Do we need Threading?

File: source/Cosmos.System2/Graphics/FullScreenCanvas.cs [hunk 3]
```diff
@@ -66,7 +66,7 @@ private static Canvas GetVideoDriver()
             {
                 return new SVGAIICanvas();
             }
-            else if (BGAExists() || PCI.GetDevice((VendorID)0x80EE, (DeviceID)0xBEEF) != null || Core.VBE.IsAvailable())
+            else if (BGAExists() || PCI.Exists(VendorID.VirtualBox, DeviceID.VBVGA) || PCI.Exists(VendorID.Bochs, DeviceID.BGA) || Core.VBE.IsAvailable())
```
Comment (quajak) [inline source/Cosmos.System2/Graphics/FullScreenCanvas.cs:69]: Can we combine all of these into a single VBECanvasAvailable?

Comment (valentinbreiz) [reply source/Cosmos.System2/Graphics/FullScreenCanvas.cs:69]: Yeah, I thought I could do that

File: source/Cosmos.HAL2/PCIDevice.cs (hunk 2 above)
Comment (valentinbreiz) [reply source/Cosmos.HAL2/PCIDevice.cs:8]: no

File: source/Cosmos.HAL2/Drivers/Video/VBEDriver.cs (hunk 1 above)
Comment (valentinbreiz) [reply source/Cosmos.HAL2/Drivers/Video/VBEDriver.cs:87]: Okay

**Pull request merged**